app/controllers/admin/passwords_controller.rb

```diff
@@ -4,4 +4,44 @@ class Admin::PasswordsController < Devise::PasswordsController
   before_action :user_logout_required
   skip_before_action :store_location
   layout "session"
+
+  before_action :check_if_totp_required, only: [:edit]
+  before_action :verify_otp_code, only: [:update]
```

Comment:
If I add the methods, I get `W: Lint/UselessMethodDefinition: Useless method definition detected.`. If I don't add the methods, I get `C: Rails/LexicallyScopedActionFilter: 'edit' is not explicitly defined on the class.`. I think I'll just leave it